splunk-otel-android/src/main/java/com/splunk/rum/SpanFilterBuilder.java

```diff
@@ -103,6 +138,25 @@ public SpanFilterBuilder rejectSpansByName(Predicate<String> spanNamePredicate)
         return replaceSpanAttribute(attributeKey, old -> attributeValuePredicate.test(old) ? null : old);
     }
 
+    /**
+     * Modify span data before it enters the exporter pipeline.
+     * <p>
+     * The value of the {@code http.url} attribute matching the passed {@code pattern} will be
+     * modified in a way that replaces all subsequences that match {@code pattern} with the
+     * {@code replacement} string. The behavior of this method is consistent with the
+     * {@link Matcher#replaceAll(String)} method.
+     *
+     * @param pattern     A regular expression pattern that matches {@code http.url} attribute
+     *                    values that should undergo replacement.
+     * @param replacement The replacement string.
+     * @return {@code this}.
+     */
+    public <T> SpanFilterBuilder replaceHttpUrl(Pattern pattern, String replacement) {
```

Comment:
The `pattern` and the `replacement` here need to be quite carefully crafted to work. I think we should at *least* put an example in the javadoc of what would work, otherwise I think that users will have a hard time figuring it out without a lot of digging.